services/html-help-service.ts

```diff
@@ -0,0 +1,161 @@
+				.replace(HtmlHelpService.RELATIVE_PATH_TO_STYLES_CSS_PLACEHOLDER, path.relative(path.dirname(filePath), this.pathToStylesCss));
+
+			this.$fs.writeFile(filePath, outputHtml).wait();
+			this.$logger.trace("Finished writing file '%s'.", filePath);
+		}).future<void>()();
+	}
+
+	public openHelpForCommandInBrowser(commandName: string): IFuture<void> {
+		return ((): void => {
+			var htmlPage = this.convertCommandNameToFileName(commandName) + HtmlHelpService.HTML_FILE_EXTENSION;
+			this.$logger.trace("Opening help for command '%s'. FileName is '%s'.", commandName, htmlPage);
+			if(!this.tryOpeningSelectedPage(htmlPage)) {
+				// HTML pages may have been skipped on post-install, lets generate them.
+				this.$logger.trace("Required HTML file '%s' is missing. Let's try generating HTML files and see if we'll find it.", htmlPage);
+				this.generateHtmlPages().wait();
+				if(!this.tryOpeningSelectedPage(htmlPage)) {
+					this.$errors.failWithoutHelp("Unable to find help for '%s'", commandName);
+				}
+			}
+		}).future<void>()();
+	}
+
+	private convertCommandNameToFileName(commandName: string): string {
+		var defaultCommandMatch = commandName.match(/(\w+?)\|\*/);
+		if(defaultCommandMatch) {
+			this.$logger.trace("Default command found. Replace current command name '%s' with '%s'.", commandName, defaultCommandMatch[1]);
+			commandName = defaultCommandMatch[1];
+		}
+
+		var availableCommands = this.$injector.getRegisteredCommandsNames(false).sort();
+		this.$logger.trace("List of registered commands: %s", availableCommands.join(", "));
+		if(commandName && !_.contains(availableCommands, commandName)) {
+			this.$errors.failWithoutHelp("Unknown command '%s'. Try '$ %s help' for a full list of supported commands.", commandName, this.$staticConfig.CLIENT_NAME.toLowerCase());
+		}
+
+		return commandName.replace(/\|/g, "-") || "index";
+	}
+	
+	private tryOpeningSelectedPage(htmlPage: string): boolean {
+		var fileList = this.$fs.enumerateFilesInDirectorySync(this.pathToHtmlPages);
+		this.$logger.trace("File list: " + fileList);
+		var pageToOpen = _.find(fileList, file => path.basename(file) === htmlPage);
+		
+		if(pageToOpen) {
+			this.$logger.trace("Found page to open: '%s'", pageToOpen);
+			this.$opener.open(pageToOpen);
+			return true;
+		} 
+		
+		this.$logger.trace("Unable to find file: '%s'", htmlPage);
+		return false;
+	}
+
+	private readManFileForCommand(commandName: string): IFuture<string> {
+		return ((): string => {
+			var mdFileName = this.convertCommandNameToFileName(commandName) + HtmlHelpService.MARKDOWN_FILE_EXTENSION;
+			this.$logger.trace("Reading help for command '%s'. FileName is '%s'.", commandName, mdFileName);
+			var helpText: string;
+
+			var markdownFile = _.find(this.$fs.enumerateFilesInDirectorySync(this.pathToManPages), file => path.basename(file) === mdFileName);
+			if(markdownFile) {
+				helpText = this.$fs.readText(markdownFile).wait();
+			} else {
```

Comment:
`else` is unneeded. Just `if(markdownFile) {
				return this.$fs.readText(markdownFile).wait(); 
}
this.$errors.failWithoutHelp("Unknown command '%s'. Try '$ %s help' for a full list of supported commands.", mdFileName.replace(".md", ""), this.$staticConfig.CLIENT_NAME.toLowerCase());
`